File: LLM-Translations/GPT-4/codenet/Go/Python/s632215495.py

class Node:
    def __init__(self, key):
        self.key = key
        self.parent = None
        self.left = None
        self.right = None

class Tree:
    def __init__(self):
        self.root = None

    def insert(self, key):
        if self.root is None:
            self.root = Node(key)
            return

        currentNode = self.root
        prevNode = None
        while currentNode is not None:
            prevNode = currentNode
            if currentNode.key < key:
                currentNode = currentNode.right
            else:
                currentNode = currentNode.left

        newNode = Node(key)
        newNode.parent = prevNode
        if prevNode.key < key:
            prevNode.right = newNode
        else:
            prevNode.left = newNode

    def print_tree(self):
        acc = []
        if self.root is None:
            print("")
            print("")
        else:
            print(" " + " ".join([str(x) for x in visit_inorder(self.root, acc)]))
            print(" " + " ".join([str(x) for x in visit_preorder(self.root, acc)]))

def visit_preorder(node, acc):
    if node is None:
        return acc
    acc.append(node.key)
    if node.left is not None:
        acc = visit_preorder(node.left, acc)
    if node.right is not None:
        acc = visit_preorder(node.right, acc)
    return acc

def visit_inorder(node, acc):
    if node is None:
        return acc
    if node.left is not None:
        acc = visit_inorder(node.left, acc)
    acc.append(node.key)
    if node.right is not None:
        acc = visit_inorder(node.right, acc)
    return acc

def main():
    n = int(input())
    tree = Tree()

    for _ in range(n):
        command = input().split()
        if command[0] == "insert":
            key = int(command[1])
            tree.insert(key)
        elif command[0] == "print":
            tree.print_tree()
        else:
            raise ValueError("unknown command: " + command[0])

if __name__ == "__main__":
    main()

File: LLM-Translations/GPT-4/codenet/Go/C/s632215495.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

typedef struct node {
    int key;
    struct node* parent;
    struct node* left;
    struct node* right;
} node;

typedef struct tree {
    node* root;
} tree;

void insert(tree* t, int key) {
    if (t->root == NULL) {
        t->root = (node*)malloc(sizeof(node));
        t->root->key = key;
        t->root->parent = NULL;
        t->root->left = NULL;
        t->root->right = NULL;
        return;
    }

    node* currentNode;
    node* prevNode;
    currentNode = t->root;
    while (currentNode != NULL) {
        prevNode = currentNode;
        if (currentNode->key < key) {
            currentNode = currentNode->right;
        } else {
            currentNode = currentNode->left;
        }
    }

    node* newNode = (node*)malloc(sizeof(node));
    newNode->key = key;
    newNode->parent = prevNode;
    newNode->left = NULL;
    newNode->right = NULL;
    if (prevNode->key < key) {
        prevNode->right = newNode;
    } else {
        prevNode->left = newNode;
    }
}

void visitPreorder(node* n, int* acc, int* size) {
    if (n == NULL) {
        return;
    }
    acc[*size] = n->key;
    (*size)++;
    if (n->left != NULL) {
        visitPreorder(n->left, acc, size);
    }
    if (n->right != NULL) {
        visitPreorder(n->right, acc, size);
    }
}

void visitInorder(node* n, int* acc, int* size) {
    if (n == NULL) {
        return;
    }
    if (n->left != NULL) {
        visitInorder(n->left, acc, size);
    }
    acc[*size] = n->key;
    (*size)++;
    if (n->right != NULL) {
        visitInorder(n->right, acc, size);
    }
}

void print(tree t) {
    int acc[100];
    int size = 0;
    visitInorder(t.root, acc, &size);
    for (int i = 0; i < size; i++) {
        printf(" %d", acc[i]);
    }
    printf("\n");
    
    size = 0;
    visitPreorder(t.root, acc, &size);
    for (int i = 0; i < size; i++) {
        printf(" %d", acc[i]);
    }
    printf("\n");
}

int main() {
    int n;
    scanf("%d", &n);

    tree t;
    t.root = NULL;
    for (int i = 0; i < n; i++) {
        char command[10];
        scanf("%s", command);
        if (strcmp(command, "insert") == 0) {
            int key;
            scanf("%d", &key);
            insert(&t, key);
        } else if (strcmp(command, "print") == 0) {
            print(t);
        } else {
            printf("unknown command: %s\n", command);
            exit(1);
        }
    }

    return 0;
}

// 
